### PR TITLE
Preview tour items

### PR DESCRIPTION
**Pull request by jonw-cogapp**
- **Pivotal ticket**: [#186171954](https://www.pivotaltracker.com/story/show/186171954)

# What does this Pull Request do?

This adds a biggish restructure and creates a new "view" tour items that have been added from search results.

It contains two other refactors:
- `useFetchItems` hook is no more. This might be more useful if we implement a search as you type feature, but it acts as a hindrance with a search button.
- `Map()` data type changes to `Array` for `tourItems`. The map's initial benefit of making data easily accessible via the ID keys was outweighed by so many other operations requiring a conversion to an array first. Lesson learned.

A big part of this is also handling a11y, and especially moving focus when the remove button removes itself.
    

# Workflow

- [ ] Before merging: After all review changes I've synchronised `develop` and rebased this branch on top of it

### CODE REVIEW

File: src/components/NavPage.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const { id, children } = props;
+  const { activeNavPage } = useContext(AppContext);
+  return (
+    <div
+      tabIndex="0"
+      id={`aic-ct-nav-page-${id}`}
+      aria-labelledby={`aic-ct-nav-button-${id}`}
+      aria-hidden={activeNavPage !== id}
+      style={activeNavPage !== id ? { display: "none" } : {}}
+    >
```
Comment (hannahb-cogapp) [inline src/components/NavPage.jsx:15]: Look into using hidden attribute

File: src/components/TourItems.jsx [hunk 2]
```diff
@@ -1,28 +1,59 @@
...
+  const [shouldAssignFocus, setShouldAssignFocus] = useState({
+    flag: false,
+    id: null,
+  });
+  const [removeButtons, setRemoveButtons] = useState([]);
 
-  window.tourItems = tourItems;
+  // When tourItems change check if this item was removed
+  // Focus will always need to be reassined in some way in that instance
+  useEffect(() => {
```
Comment (hannahb-cogapp) [inline src/components/TourItems.jsx:18]: Please correct the typo

File: src/contexts/AppContext.jsx [hunk 3]
```diff
@@ -8,17 +8,20 @@ export const AppContext = createContext();
...
+  const [navPages, setNavPages] = useState(navPagesValue || []);
+  const [activeNavPage, setActiveNavPage] = useState(0);
   const [tourItems, tourItemsDispatch] = useReducer(
     tourItemsReducer,
-    new Map(
-      tourItemsValue ? tourItemsValue.map((item) => [item.id, item]) : [],
-    ),
+    tourItemsValue || [],
   );
+  const navSearchButtonRef = useRef(null);
```
Comment (hannahb-cogapp) [inline src/contexts/AppContext.jsx:25]: Check if this can be moved out of AppContext

**Review by hannahb-cogapp — APPROVED** [review]
A couple of things to look into, but otherwise looks good

File: src/components/NavPage.jsx (hunk 1 above)
Comment (jonw-cogapp) [reply src/components/NavPage.jsx:15]: Had a look at this, after reading [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) and [CSSTricks](https://css-tricks.com/the-hidden-attribute-is-visibly-weak/) I feel like I understand this a little better, and it's a nice trick... But that I don't feel it's worth changing in this case

Reasons being:
- I suspect we'll land on different CSS later
- Being easily overrideable It seems possible `hidden` may show the element when we don't expect it to
- Not transitionable (neither is `display:none`, but using different CSS is less of a leap than not using any CSS)

File: src/contexts/AppContext.jsx (hunk 3 above)
Comment (jonw-cogapp) [reply src/contexts/AppContext.jsx:25]: Gonna leave this in `AppContext` as it gets set in `<Navigation/>` and read in `<TourItems/>`, but there isn't a direct hierarchy that could pass this down apart from the `AppContext`. Looking at this now it seems logical

**Pull request merged**